libcontainer/intelrdt/intelrdt.go

```diff
@@ -744,16 +837,79 @@ func (m *intelRdtManager) Set(container *configs.Config) error {
 	return nil
 }
 
-func (raw *intelRdtData) join(id string) (string, error) {
-	path := filepath.Join(raw.root, id)
-	if err := os.MkdirAll(path, 0755); err != nil {
-		return "", NewLastCmdError(err)
+func switchGroup(m *intelRdtManager, GetIntelRdtPath func(string, string) (string, error)) error {
```

Comment:
Based on the comments above, this could be simplified to:
`func switchGroup(id string, grouptype string) error {}`